cmake/onnx.cmake

```diff
@@ -0,0 +1,34 @@
+PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS
```

Comment:
I am not sure if this is the convention, but usually files under cmake/ are utility files that cmake uses for custom builds. For the content of this file, it seems to be a better fit as a onnx/CMakeLists.txt file. Does that make sense?

Reply:
Hmmm, I think it in this way, 1) put all cmake related files into /cmake folder, instead of having them in different folders. 2) we may have more than one *.cmake later for different project separations, Sounds good?

Reply:
OK, now I merged things in one CmakeLists.txt and put it in root folder, which resolved several of the comments here.